version: "3.8"

services: 
    barata-api:
        build: 
            context: .
            dockerfile: Dockerfile.dev
        container_name: barata-api
        command: python3 manage.py runserver 0.0.0.0:8000
        volumes: 
            - .:/app
        # ports:
        #     - 8000:8000
        env_file: 
            - .env
        restart: "on-failure"
        depends_on: 
            - barata-db
        networks: 
            - barata-network
    
    barata-db:
        image: kartoza/postgis:13-3.1
        ports: 
            - 5432:5432
        volumes: 
            - db-volume:/var/lib/postgresql
        environment: 
            - POSTGRES_USER=${POSTGRES_USER}
            - POSTGRES_PASS=${POSTGRES_PASS}
            - POSTGRES_DBNAME=${POSTGRES_DBNAME}
        container_name: barata-db
        restart: "on-failure"
        networks: 
            - barata-network
    
    barata-client:
        build:
            context: ./barata_client
            dockerfile: Dockerfile.dev
        container_name: barata-client
        volumes: 
            - ./app/node_modules
            - ./barata_client:/app
        depends_on: 
            - barata-api
        networks: 
            - barata-network
            
    barata-nginx:
        restart: always
        depends_on: 
            - barata-client
        build:
            context: ./nginx
            dockerfile: Dockerfile.dev
        container_name: barata-nginx
        ports: 
            - 8080:80
        networks: 
            - barata-network

networks: 
    barata-network:
        driver: bridge
        name: barata-network

volumes: 
    db-volume:
        name: db-volume